packages/lisk-p2p/src/sanitization.ts

```diff
@@ -92,3 +98,32 @@ export const sanitizePeerInfoList = (
 		throw new InvalidRPCResponse('Invalid response type');
 	}
 };
+
+export const validateRPCRequest = (request: unknown): P2PRequest => {
+	if (!request) {
+		throw new InvalidRPCRequestError('Invalid request');
+	}
+
+	const rpcRequest = request as P2PRequest;
+	if (typeof rpcRequest.procedure !== 'string') {
+		throw new InvalidRPCRequestError('Request procedure name is not a string');
+	}
+	if (typeof rpcRequest.data !== 'object') {
+		throw new InvalidRPCRequestError('Invalid request data');
+	}
+
+	return rpcRequest;
+};
+
+export const sanitizeProtocolMessage = (message: unknown): ProtocolMessagePacket => {
```

Comment:
actually before, all the functions were prefixed as process or validate. But later @jondubois renamed all of them to sanitize including the file name because these functions are not only validating but type casting or sanitizing it to some protocol type or preferred data structure.

Reply:
Mmm we have similar functionality in transaction with validating and the checking so maybe it’s better to standardize?

Reply:
Okay then I think it makes sense to just use `validate` here too  👍